spectree/plugins/starlette_plugin.py

```diff
@@ -4,13 +4,20 @@
 from json import JSONDecodeError
 
 from pydantic import ValidationError
+from starlette.responses import JSONResponse
 
 from .base import BasePlugin, Context
 
 METHODS = {"get", "post", "put", "patch", "delete"}
 Route = namedtuple("Route", ["path", "methods", "func"])
 
 
+class PydanticResponse(JSONResponse):
+    def render(self, content) -> bytes:
+        self._model_class = content.__class__
+        return super().render(content.dict())
+
+
```

Comment:
I see your point. This will require `starlette` to be installed even for the `flask` and `falcon` users, which is not reasonable. Also makes `starlette_plugin` special.

I don't have a very good solution for now.

1. just don't support this feature for starlette
2. user needs to wrap it with something like the `PydanticResonse`
3. hack and change the behavior of `JSONResponse` when init `StarlettePlugin`
4. use `PlainTextResponse` then convert it to `JSONResponse` internally
5. add a new method to `SpecTree` to process the `BaseModel` response, similar to what your proposal `spec.convert_response(model: BaseModel, ...)`

Reply:
Is there an issue with just having it in the backend as it is now and doing `from spectree.plugins.startlette_plugin import PydanticResponse`?

Reply:
The main issue is that `flask` and `falcon` users will have to install `starlette`. Otherwise, it will raise

```sh
Traceback (most recent call last):
  File "examples/flask_demo.py", line 8, in <module>
    from spectree import Response, SpecTree
  File "/home/user/GitHub/spectree/spectree/__init__.py", line 5, in <module>
    from .spec import SpecTree
  File "/home/user/GitHub/spectree/spectree/spec.py", line 9, in <module>
    from .plugins import PLUGINS, BasePlugin
  File "/home/user/GitHub/spectree/spectree/plugins/__init__.py", line 4, in <module>
    from .starlette_plugin import StarlettePlugin
  File "/home/user/GitHub/spectree/spectree/plugins/starlette_plugin.py", line 7, in <module>
    from starlette.responses import JSONResponse
ModuleNotFoundError: No module named 'starlette'
```

Reply:
Ah sorry, never realised all backends were imported in the `__init__.py`, I've changed this to a function that lazily imports from starlette, a little hacky but it works.